includes/customer-functions.php

```diff
@@ -636,14 +636,14 @@ function edd_maybe_add_customer_address( $customer_id = 0, $data = array() ) {
 		'region'      => '',
 		'country'     => '',
 		'postal_code' => '',
-		'type'        => 'billing',
```

Comment:
For the empty address check, we need to make sure that only address elements are included in that array--if a `type` is set in the `$data`, but nothing else, and we're checking that for emptiness, it will create an otherwise empty address. Confirmed with @ashleyfae; updating the comment above to clarify.